### DISCUSSION TITLE
Reverse Proxies - Websockets, etc

### DISCUSSION THREAD

**Discussion by kevdogg** [emqx/emqx#13747] [General]
I'm a little new to websocket architecture but not setting up nginx/traefik reverse proxies.  I read the documentation regarding nginx reverse proxies along with the nginx configuration for setting up wss. [https://docs.emqx.com/en/emqx/latest/deploy/cluster/lb-nginx.html]( https://docs.emqx.com/en/emqx/latest/deploy/cluster/lb-nginx.html)

I guess I'm just a little confused.  I'm aware web sockets on a channel whereby clients/server can "talk" to each other similar to a local socket, however I'm just to wondering in what application with emqx would this be used?  What typical clients would access emqx using websockets. Right now I'm using emqx as my mqtt broker in bewteen z2m and Home Assistant.  

In terms of the nginx configuration -- do you need to setup http proxy and wss proxy by combining the two configurations?  The basic http/https proxy just has a server block, whereas the websocket location block has /mqtt as the location.  Do I need both of these.  

I'm sure these are very basic questions and hopefully I'm asking these appropriately.

**Comment by zmstone** [comment]
Webhook is useful when implementing MQTT client in web applications running in a browser so they can interact with EMQX over http/https protocol.
I don't think you'll need to expose the websocket endpoint if your clients connects EMQX over TCP or TLS (port 1883).
